### ISSUE TITLE
[FEAT]: Sync only one of the components of a profile

### ISSUE THREAD

**Issue by 1ku1** [4Source/settings-profiles-obsidian-plugin#60]
### Description of the unfavorable solution

I changed hotkeys, so I want to trigger the sync of **hotkey component** of the used profile (without having to create a different profile and manage switching between the two).

### Describe the feature you would like to have

May be a save/reload button in the profile options page or through a subcommand(?)

**Comment by 4Source** [comment]
@1ku1 I created a beta release containing some new command could you try it.

[0.5.10-beta.0](https://github.com/4Source/settings-profiles-obsidian-plugin/releases/tag/0.5.10-beta.0)

You could install it via a plugin of mine [VARE](https://obsidian.md/plugins?id=vare) or add the release files manually 

I am waiting for your feedback hop this fit your needs

**Comment by 1ku1** [comment]
@4Source 

Thanks. I tried it. It seems to work correctly and I have two notes:

1. There is no reload partially. For example, only import hotkeys.
2. There isn't an easy way to set a profile as default without applying it (reloading from it).
  - Create a test `Vault1`, `Vault2`, make changes to `Vault2` only
  - Create settings profile `Test`.
  - Set Test as default for `Vault1` , save to it.
  - Set it as default for `Vault2`. Now your changes in `Vault2` are overridden because the profile has to be reloaded.

**Comment by 4Source** [comment]
> 1. There is no reload partially. For example, only import hotkeys.

Had this but not function properly this requires some struktural changes I try to add it.

> There isn't an easy way to set a profile as default without applying it (reloading from it).
> * Create a test `Vault1`, `Vault2`, make changes to `Vault2` only
> * Create settings profile `Test`.
> * Set Test as default for `Vault1` , save to it.
> * Set it as default for `Vault2`. Now your changes in `Vault2` are overridden because the profile has to be reloaded.

What your are calling default is the selection of the profile. I think this is the expected behavior your are switching to a profile this means you would like to have the settings in this profile to be loaded. What are you trying to achieve may there is a solution to this

**Comment by 4Source** [comment]
I created a new beta release [0.5.10-beta.2](https://github.com/4Source/settings-profiles-obsidian-plugin/releases/tag/0.5.10-beta.2) with the Load partially command and a command that allows to save the current settings to a specified profile

**Comment by 1ku1** [comment]
@4Source 
I have tried beta 2. Things seem to work correctly

>What your are calling default is the selection of the profile. I think this is the expected behavior your are switching to a profile this means you would like to have the settings in this profile to be loaded.

Not necessarily. see below

> What are you trying to achieve may there is a solution to this

For example, if you want to export only the hotkeys of Vault 2, you won't be able to run the command `save current profile partially` because hotkeys will be overwritten by setting the profile as default.

**Comment by 4Source** [comment]
If I understand you correctly you would like to save the current hotkey settings to a not selected profile.

This is possible with ``Save to profile partially`` there you could select the target profile and the settings you would like to save

**Comment by 1ku1** [comment]
> This is possible with `Save to profile partially` there you could select the target profile and the settings you would like to save

You are right, but in short, I have to be careful whenever I install and configure settings-profiles to a new vault so as not to overwrite the profile. I want to be able to set it as default one to read to/write from for this new vault then calmly examine which settings I want to import and export.

When setting up the new vault, overwriting everything default/local is kinda scary 😅

**Comment by 4Source** [comment]
Ok I think your are using the plugin completely different than I do. 

To clarify:
- You would like to be be able to switch to a profile in a new vault (or any other) without loading the settings from the profile in the vault? 
_Note: I can add this but I don't understand what your scared of by loading setting in an empty vault_ 😅
- You would like to have a message before changes are saved to a profile?

**Comment by 1ku1** [comment]
>You would like to be be able to switch to a profile in a new vault (or any other) without loading the settings from the profile in the vault?

Yes, please.

> Note: I can add this but I don't understand what your scared of by loading setting in an empty vault 😅

1. It is not empty, since I usually copy over the `.obsidian` file of the vault with the closest use case to quickly "install the plugins" and have the settings closest to what I need. I then review and move from there.
2. On the other hand, even when I create an empty vault, I don't like to overwrite the defaults without review first.

>You would like to have a message before changes are saved to a profile?

I think that would be helpful. You can always add a checkbox "Don't ask again".

**Comment by 4Source** [comment]
> 1. It is not empty, since I usually copy over the `.obsidian` file of the vault with the closest use case to quickly "install the plugins" and have the settings closest to what I need. I then review and move from there.

Why you don't let this do this plugin you just need to install this one and than you can just install them by selecting profile

Ok I will work on this

**Comment by 1ku1** [comment]
> Why you don't let this do this plugin you just need to install this one and than you can just install them by selecting profile

I plan to. Once I can tell it which settings and plugins to ignore :D

> Ok I will work on this

Thanks a lot

**Comment by 4Source** [comment]
![image](https://github.com/user-attachments/assets/d186ecae-4b23-4cef-bb0e-e93e7ea567a7)

The profile switcher now allows to switch without loading

[0.5.10-beta.3](https://github.com/4Source/settings-profiles-obsidian-plugin/releases/tag/0.5.10-beta.3)

**Comment by 4Source** [comment]
Could you test the new Commands, are there working? I will also do more testing and the I will merge it.

**Comment by 1ku1** [comment]
I tried it and the new command is working (to the extent of my testing). I have a small suggestion: to have the same option in the settings page: in a new clean vault > settings-profile settings page > when you click switch to profile: maybe show a small drop down menu with:
- Select
- Select and apply

Thanks
